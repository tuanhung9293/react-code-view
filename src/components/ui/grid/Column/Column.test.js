import React from 'react'
import {shallow} from 'enzyme'
import {Column} from './'

it('renders children', () => {
  expect(shallow(
    <Column>
      <p>I feel like a child</p>
    </Column>
  )).toMatchSnapshot()
})

it('creates a grid column when it takes a span parameter from 1 to 12', () => {
  for (let i = 1; i <= 12; i++) {
    expect(shallow(
      <Column span={i}>
        <p>I feel like a child</p>
      </Column>
    )).toMatchSnapshot()
  }
})

it('takes breakpoint to alter column span according to screen width', () => {
  expect(shallow(
    <Column span="6" breakPoints={[{maxWidth: 600, span: 12}]} windowWidth={500} />
  )).toMatchSnapshot()
})

it('hides element if span is 0', () => {
  expect(shallow(<Column span="0" />)).toMatchSnapshot()
  expect(shallow(
    <Column span="6" breakPoints={[{maxWidth: 600, span: 0}]} windowWidth={500} />
)).toMatchSnapshot()
})