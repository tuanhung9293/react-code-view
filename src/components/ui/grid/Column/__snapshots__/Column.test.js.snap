// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a grid column when it takes a span parameter from 1 to 12 1`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "8.333333333333334%",
      "flexGrow": 0,
      "maxWidth": "8.333333333333334%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "8.333333333333334%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 2`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "16.666666666666668%",
      "flexGrow": 0,
      "maxWidth": "16.666666666666668%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "16.666666666666668%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 3`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "25%",
      "flexGrow": 0,
      "maxWidth": "25%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "25%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 4`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "33.333333333333336%",
      "flexGrow": 0,
      "maxWidth": "33.333333333333336%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "33.333333333333336%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 5`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "41.66666666666667%",
      "flexGrow": 0,
      "maxWidth": "41.66666666666667%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "41.66666666666667%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 6`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "50%",
      "flexGrow": 0,
      "maxWidth": "50%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "50%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 7`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "58.333333333333336%",
      "flexGrow": 0,
      "maxWidth": "58.333333333333336%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "58.333333333333336%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 8`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "66.66666666666667%",
      "flexGrow": 0,
      "maxWidth": "66.66666666666667%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "66.66666666666667%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 9`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "75%",
      "flexGrow": 0,
      "maxWidth": "75%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "75%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 10`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "83.33333333333334%",
      "flexGrow": 0,
      "maxWidth": "83.33333333333334%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "83.33333333333334%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 11`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "91.66666666666667%",
      "flexGrow": 0,
      "maxWidth": "91.66666666666667%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "91.66666666666667%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`creates a grid column when it takes a span parameter from 1 to 12 12`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "100%",
      "flexGrow": 0,
      "maxWidth": "100%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "100%",
    }
  }
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`hides element if span is 0 1`] = `
<div
  className="base grid"
  style={
    Object {
      "display": "none",
    }
  }
/>
`;

exports[`hides element if span is 0 2`] = `
<div
  className="base grid"
  style={
    Object {
      "display": "none",
    }
  }
/>
`;

exports[`renders children 1`] = `
<div
  className="base column"
  style={null}
>
  <p>
    I feel like a child
  </p>
</div>
`;

exports[`takes breakpoint to alter column span according to screen width 1`] = `
<div
  className="base grid"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "flexBasis": "100%",
      "flexGrow": 0,
      "maxWidth": "100%",
      "msFlexPositive": 0,
      "msFlexPreferredSize": "100%",
    }
  }
/>
`;
